option java_package = "com.mogujie.tt.protobuf";
//option java_outer_classname = "MOGUJIEIMMessage";
option optimize_for = LITE_RUNTIME;

// service id
enum ServiceID{
    SID_LOGIN           = 0x0001;           // for login
    SID_BUDDY_LIST      = 0x0002;		    // for friend list
    SID_MSG             = 0x0003;           //
    SID_GROUP           = 0x0004;           // for group message
    SID_FILE            = 0x0005;
    SID_SWITCH_SERVICE  = 0x0006;
    SID_OTHER           = 0x0007;
    SID_INTERNAL        = 0x0008;		     
}

// command id for login
enum LoginCmdID{
    CID_LOGIN_REQ_MSGSERVER         = 0x0101; 	//
    CID_LOGIN_RES_MSGSERVER         = 0x0102;	//
    CID_LOGIN_REQ_USERLOGIN         = 0x0103;	//
    CID_LOGIN_RES_USERLOGIN         = 0x0104;	//
    CID_LOGIN_REQ_LOGINOUT          = 0x0105;	//
    CID_LOGIN_RES_LOGINOUT          = 0x0106; 	//
    CID_LOGIN_KICK_USER				= 0x0107; 	//
    CID_LOGIN_REQ_DEVICETOKEN       = 0x0108;  //
    CID_LOGIN_RES_DEVICETOKEN       = 0x0109;  //
    CID_LOGIN_REQ_KICKPCCLIENT      = 0x010a;
    CID_LOGIN_RES_KICKPCCLIENT      = 0x010b;
    CID_LOGIN_REQ_PUSH_SHIELD       = 0x010c;	//勿扰
    CID_LOGIN_RES_PUSH_SHIELD       = 0x010d; 	//
    CID_LOGIN_REQ_QUERY_PUSH_SHIELD = 0x010e; 	//
    CID_LOGIN_RES_QUERY_PUSH_SHIELD = 0x010f;
}

// command id for msg
enum MessageCmdID{
    CID_MSG_DATA					= 0x0301;	//
    CID_MSG_DATA_ACK				= 0x0302; 	//
    CID_MSG_READ_ACK				= 0x0303; 	//
    CID_MSG_READ_NOTIFY     		= 0x0304;    //  已读消息通知
    CID_MSG_TIME_REQUEST 			= 0x0305;	//
    CID_MSG_TIME_RESPONSE			= 0x0306; 	//
    CID_MSG_UNREAD_CNT_REQUEST		= 0x0307; 	//
    CID_MSG_UNREAD_CNT_RESPONSE		= 0x0308;	//
    CID_MSG_LIST_REQUEST            = 0x0309;    //获取指定队列消息
    CID_MSG_LIST_RESPONSE           = 0x030a;
    CID_MSG_GET_LATEST_MSG_ID_REQ   = 0x030b;
    CID_MSG_GET_LATEST_MSG_ID_RSP   = 0x030c;
    CID_MSG_GET_BY_MSG_ID_REQ       = 0x030d;
    CID_MSG_GET_BY_MSG_ID_RES       = 0x030e;
    CID_MSG_GET_DAYS_REQ            = 0x0341;
    CID_MSG_GET_DAYS_RES            = 0x0342;
    CID_MSG_GET_DURING_TIME_REQ     = 0x0343;
    CID_MSG_GET_DURING_TIME_RES     = 0x0344;
}

enum UserStatType{
	USER_STATUS_ONLINE 	= 1;
	USER_STATUS_OFFLINE	= 2;
	USER_STATUS_LEAVE	= 3;

}

enum ClientType{
	CLIENT_TYPE_WINDOWS     = 0x01;
    CLIENT_TYPE_MAC         = 0x02;
    CLIENT_TYPE_IOS         = 0x11;
    CLIENT_TYPE_ANDROID     = 0x12;
    CLIENT_TYPE_WEB         = 0x21;
}

enum ResultType{
	REFUSE_REASON_NONE				= 0;
	REFUSE_REASON_NO_MSG_SERVER		= 1;
	REFUSE_REASON_MSG_SERVER_FULL 	= 2;
	REFUSE_REASON_NO_DB_SERVER		= 3;
	REFUSE_REASON_NO_LOGIN_SERVER	= 4;
	REFUSE_REASON_NO_ROUTE_SERVER	= 5;
	REFUSE_REASON_DB_VALIDATE_FAILED = 6;
	REFUSE_REASON_VERSION_TOO_OLD	= 7;

}

enum MsgType{
	MSG_TYPE_SINGLE_TEXT    = 0x01;
    MSG_TYPE_SINGLE_AUDIO   = 0x02;
    MSG_TYPE_GROUP_TEXT     = 0x11;
    MSG_TYPE_GROUP_AUDIO    = 0x12;
}

enum SessionType{
	SESSION_TYPE_SINGLE = 1;          	//单个用户会话
	SESSION_TYPE_GROUP = 2;          	//群会话
}

message UserInfo{
	required uint32 user_id = 1;
	required uint32 user_gender = 2; 	//// 用户性别,男：1 女：2 人妖/外星人：0
	required string user_nick_name = 3;	//绰号
	required string avatar_url = 4;
	required uint32 department_id = 5;
	required string email = 6;
	required string user_real_name = 7;	//真名
	required string user_tel = 8;
	required string user_domain = 9;	//用户名拼音
	required uint32 status = 10;        //0:在职  1. 试用期 2. 正式 3. 离职 4.实习,  client端需要对“离职”进行不展示
	required uint32 user_type = 11;
	optional string sign_info = 12;
}

message MsgInfo{
        required uint32 msg_id = 1;
        required uint32 from_session_id = 2;   //发送的用户id
        required uint32 create_time = 3;
        required MsgType msg_type = 4;
        required bytes msg_data = 5;
}

message IMLoginReq{
//service id:	0x0001
	//cmd id:		0x0103
	required string user_name = 1;
	required string password = 2;
	required UserStatType online_status = 3;
	required ClientType client_type = 4;
	optional string client_version = 5;
	optional uint32 crypto_flag = 6;
}

message IMLoginRes{
	//cmd id:		0x0104
	required uint32 server_time = 1;
	required ResultType result_code = 2;
	optional string result_string = 3;
	optional UserStatType online_status = 4;
	optional UserInfo user_info = 5;
	optional string client_key = 6;
}

message IMMsgData{
//service id  0x0003
	//cmd id:		0x0301
	required uint32 from_user_id = 1;				//消息发送方
	required uint32 to_session_id = 2;				//消息接受方
	required uint32 msg_id = 3;
	required uint32 create_time = 4; 
	required MsgType msg_type = 5;
	required bytes msg_data = 6;
	optional string token = 7;
	optional uint32 crypto_flag = 8;
	optional uint32 forwarded = 9;
	optional bytes attach_data = 20;
}

message IMMsgDataAck{
//service id  0x0003
	//cmd id:		0x0302
	required uint32 user_id = 1;			//发送此信令的用户id
	required uint32 session_id = 2;				
	required uint32 msg_id = 3;
	required SessionType session_type = 4;
	optional string token = 5;
	optional string create_time = 6;
	optional bytes attach_data = 20;
}

message IMGetTimeMsgListReq{
        //cmd id:               0x0343
        required uint32 user_id = 1;
        required SessionType session_type = 2;
        required uint32 session_id = 3;
        required uint32 start_time = 4;
        required uint32 end_time = 5;
        required uint32 offset = 6;
        required uint32 count = 7;
        optional bytes attach_data = 20;
}

message IMGetTimeMsgListRsp{
        //cmd id:               0x0344
        required uint32 user_id = 1;
        required SessionType session_type = 2;
        required uint32 session_id = 3;
        repeated MsgInfo msg_list = 5;
        optional bytes attach_data = 20;
}
